js/src/carousel.js

```diff
@@ -405,18 +406,20 @@ class Carousel extends BaseComponent {
 
   _setActiveIndicatorElement(element) {
     if (this._indicatorsElement) {
-      const indicators = SelectorEngine.find(SELECTOR_ACTIVE, this._indicatorsElement)
+      const activeIndicators = SelectorEngine.find(SELECTOR_ACTIVE, this._indicatorsElement)
 
-      for (let i = 0; i < indicators.length; i++) {
-        indicators[i].classList.remove(CLASS_NAME_ACTIVE)
+      for (let i = 0; i < activeIndicators.length; i++) {
+        activeIndicators[i].classList.remove(CLASS_NAME_ACTIVE)
+        activeIndicators[i].removeAttribute('aria-current')
       }
 
-      const nextIndicator = this._indicatorsElement.children[
-        this._getItemIndex(element)
-      ]
+      const indicators = SelectorEngine.find(SELECTOR_INDICATOR, this._indicatorsElement)
 
-      if (nextIndicator) {
-        nextIndicator.classList.add(CLASS_NAME_ACTIVE)
+      for (let i = 0; i < indicators.length; i++) {
+        if (Number.parseInt(indicators[i].getAttribute('data-bs-slide-to'), 10) === this._getItemIndex(element)) {
+          indicators[i].classList.add(CLASS_NAME_ACTIVE)
+          indicators[i].setAttribute('aria-current', 'true')
+        }
```

Comment:
Not sure why you are iterating over all the buttons, you can directly get the target indicator.

Reply:
my JS is barely passable, that's why ;)

Reply:
if you have a cleaner way of targeting the indicator whose `data-bs-slide-to` matches `this._getItemIndex(element)`, happy to change this

Reply:
Since the indecators are placed in the same order as the slides, the indicator can also be found by the index of the item (the previous implementation). But if it should be found by `data-bs-slide-to`, then my suggestion:

```suggestion
const targetIndicator = [].find.call(SelectorEngine.find(SELECTOR_INDICATOR, this._indicatorsElement), indicator => {
  return Manipulator.getDataAttribute(indicator, 'slide-to') === this._getItemIndex(element)
})

if (targetIndicator) {
  targetIndicator.classList.add(CLASS_NAME_ACTIVE)
  targetIndicator.setAttribute('aria-current', 'true')
}
```

Reply:
> Since the indecators are placed in the same order as the slides

it may or may not be the case though. I don't think we enforce that anywhere (since authors can add whichever `data-bs-slide-to` they want - and some authors may decide that they want to have it in some quirky order, or reverse order, or whatever)

Reply:
> But the previous implementation was better IMO

so, should I just leave it as is, with the `for` loop? personally, i find it at lot more readable/understandable the way I had it.

Reply:
Then, you should break the loop inside the condition at least.

Reply:
done, i think